packages/@aws-cdk-testing/framework-integ/test/aws-ssm/test/integ.parameter-store-string-sharing.ts

```diff
@@ -0,0 +1,140 @@
+import * as cdk from 'aws-cdk-lib';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import {
+  aws_ssm as ssm,
+  aws_ram as ram,
+  StackProps,
+} from 'aws-cdk-lib';
+
+/**
+ * This integration test requires two accounts with an AWS SSM Parameter created from the producing account,
+ * sharing with the consuming account, which would accept the sharing invitation and consume the shared parameter
+ * in its consuming stack.
+ *
+ * To test it locally:
+ *
+ * $ export CDK_INTEG_ACCOUNT=<AWS_ACCOUNT_PRODUCING>
+ * $ export CDK_INTEG_CROSS_ACCOUNT=<AWS_ACCOUNT_CONSUMING>
+ *
+ * And run:
+ *
+ * $ yarn integ test/aws-ssm/test/integ.parameter-store-string-sharing.js --update-on-failed
+ *
+ * The `producing` stack would be created from the sharing account in `us-east-1` followed by the `accepting` and `consuming` stacks
+ * being created from the comsuming account in the same region.
+ *
+ * `stackUpdateWorkflow` has to be disabled as the accepting stack can not re-accept the sharing invite.
+ *
+ */
+
+export interface ProducingStackProps extends StackProps {
+  readonly sharingToAccount: string;
+  readonly parameterName: string;
+}
+
+class ProducingStack extends cdk.Stack {
+  readonly resourceShareArn: string;
+  readonly parameterArn: string;
+  constructor(scope: cdk.App, id: string, props: ProducingStackProps) {
+    super(scope, id, props);
+
+    const publicParameter = new ssm.StringParameter(this, 'String', {
+      parameterName,
+      stringValue: 'Abc123',
+      tier: ssm.ParameterTier.ADVANCED,
+    });
+
+    this.parameterArn = publicParameter.parameterArn;
+
+    // create resource share
+    const share = new ram.CfnResourceShare(this, 'ParamShare', {
+      name: 'ParamShare',
+      allowExternalPrincipals: true,
+      principals: [props.sharingToAccount],
+      resourceArns: [this.parameterArn],
+      permissionArns: [
+        'arn:aws:ram::aws:permission/AWSRAMDefaultPermissionSSMParameterReadOnly',
+      ],
+    });
+
+    this.resourceShareArn = share.attrArn;
+    new cdk.CfnOutput(this, 'ResourceShareArn', { value: this.resourceShareArn });
+    new cdk.CfnOutput(this, 'ParameterArn', { value: this.parameterArn });
+
+    share.node.addDependency(publicParameter);
+
+  }
+}
+class AcceptingStack extends cdk.Stack {
```

Comment:
slightly confused as to why we have accepting stack and consuming stack as different, can we combine both into one, i.e. keep the accepting API call in consuming stack.

Reply:
We need to ensure the order:

1. producer Account to create a ssm parameter
2. producer Account to create a RAM share with the consumer Account
3. consumer Account to list/select the invitation via AwsApiCall
4. consumer Account to accept the invitation via  yet another AwsApiCall
5. consumer Account to consume this shared parameter


In the `ConsumingStack` there's only one statement

```ts
const remoteParam = ssm.StringParameter.fromStringParameterArn(this, 'RemoteParam', props.parameterArn);
```

As this is not a resource or construct, it won't be able to depend on the `AwsApiCall` which is a Construct. But we still have to wait before the invitation is accepted.

The workaround here is to group them into stacks and ensure stack dependency.

Reply:
Got it so the dependency is to make sure the invite has been accepted before consumption. Thank you for explanation.